Find a functionally equivalent code:android.bluetooth.BluetoothGattServer.connect:COMMENT
Method Modifier: public      
Comment:/**
 * Initiate a connection to a Bluetooth GATT capable device.
 *
 * <p>The connection may not be established right away, but will be
 * completed when the remote device is available. A
 * {@link BluetoothGattServerCallback#onConnectionStateChange} callback will be
 * invoked when the connection state changes as a result of this function.
 *
 * <p>The autoConnect paramter determines whether to actively connect to
 * the remote device, or rather passively scan and finalize the connection
 * when the remote device is in range/available. Generally, the first ever
 * connection to a device should be direct (autoConnect set to false) and
 * subsequent connections to known devices should be invoked with the
 * autoConnect parameter set to true.
 *
 * <p>Requires {@link android.Manifest.permission#BLUETOOTH} permission.
 *
 * @param autoConnect Whether to directly connect to the remote device (false)
 * or to automatically connect as soon as the remote
 * device becomes available (true).
 * @return true, if the connection attempt was initiated successfully
 */

@@ -4,11 +4,11 @@
     if (mService == null || mServerIf == 0)
         return false;
     try {
-        mService.serverConnect(mServerIf, device.getAddress(), // autoConnect is inverse of "isDirect"
-        autoConnect ? false : true);
+        mService.serverConnect(mServerIf, device.getAddress(), autoConnect ? false : true, // autoConnect is inverse of "isDirect"
+        mTransport);
     } catch (RemoteException e) {
         Log.e(TAG, "", e);
         return false;

